mesh/pmesh.cpp

```diff
@@ -3055,6 +3055,8 @@ bool ParMesh::NonconformingDerefinement(Array<double> &elem_error,
    long glob_size = ReduceInt(derefs.Size());
    if (!glob_size) { return false; }
 
+   DeleteFaceNbrData();
```

Comment:
It is probably better to call this at the beginning of the function, right after the `MFEM_VERIFY` checks.

Reply:
I did put it there at first but then I noticed that the method can return without preforming any de-refinement (see line 3056).  In such cases I thought it would be best to leave everything unchanged.

Reply:
OK, this makes sense. Can you add a comment to clarify this, e.g. something like
```c++
// Destroy face-neighbor data only when actually de-refining.
```